### ISSUE TITLE
Query :: orderby + filter + result operator throws

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#1620]
Repro:

```
    public class Customer
    {
        public int Id { get; set; }
    }

    public class Order
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
    }

    public class MyContext : DbContext
    {
        public DbSet<Customer> Customers { get; set; }
        public DbSet<Order> Orders { get; set; }

        protected override void OnConfiguring(DbContextOptions options)
        {
            options.UseSqlServer(new SqlConnectionStringBuilder { DataSource = ".", InitialCatalog = "Repro1620", MultipleActiveResultSets = true, IntegratedSecurity = true }.ToString());
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            using (var ctx = new MyContext())
            {
                ctx.Database.EnsureDeleted();
                ctx.Database.EnsureCreated();

                ctx.Customers.OrderBy(c => c.Id).Where(c => c.Id > 5).Any();
            }
        }
    }

```

throws:

Unhandled Exception: System.ArgumentException: Expression of type 'System.Collec
tions.Generic.IEnumerable`1[System.Data.Common.DbDataReader]' cannot be used for
 parameter of type 'System.Collections.Generic.IEnumerable`1[Microsoft.Data.Enti
ty.Query.QuerySourceScope]' of method 'System.Linq.IOrderedEnumerable`1[Microsof
t.Data.Entity.Query.QuerySourceScope] _OrderBy[QuerySourceScope,Int32](System.Co
llections.Generic.IEnumerable`1[Microsoft.Data.Entity.Query.QuerySourceScope], S
ystem.Func`2[Microsoft.Data.Entity.Query.QuerySourceScope,System.Int32], Remotio
n.Linq.Clauses.OrderingDirection)'
   at Microsoft.Data.Entity.Query.CompiledQueryCache.CompileQuery(DataStore data
Store, MethodInfo compileMethodInfo, Type resultItemType, QueryModel queryModel)
 in D:\k\EntityFramework\src\EntityFramework.Core\Query\CompiledQueryCache.cs:li
ne 341
   at Microsoft.Data.Entity.Query.CompiledQueryCache.<>c__4`1.<Execute>b__4_0(Ex
pression q, DataStore ds) in D:\k\EntityFramework\src\EntityFramework.Core\Query
\CompiledQueryCache.cs:line 65
   at Microsoft.Data.Entity.Query.CompiledQueryCache.<>c__DisplayClass7_0.<GetOr
Add>b__0(ICacheSetContext c) in D:\k\EntityFramework\src\EntityFramework.Core\Qu
ery\CompiledQueryCache.cs:line 156
   at Microsoft.Framework.Cache.Memory.CacheExtensions.<>c__DisplayClass7_0`1.<S
et>b__0(ICacheSetContext context)
   at Microsoft.Framework.Cache.Memory.MemoryCache.Set(String key, IEntryLink li
nk, Object state, Func`2 create)
   at Microsoft.Framework.Cache.Memory.CacheExtensions.Set[T](IMemoryCache cache
, String key, Object state, Func`2 create)
   at Microsoft.Framework.Cache.Memory.CacheExtensions.GetOrSet[T](IMemoryCache
cache, String key, Object state, Func`2 create)
   at Microsoft.Data.Entity.Query.CompiledQueryCache.GetOrAdd(Expression query,
QueryContext queryContext, DataStore dataStore, Boolean isAsync, Func`3 compiler
) in D:\k\EntityFramework\src\EntityFramework.Core\Query\CompiledQueryCache.cs:l
ine 148
   at Microsoft.Data.Entity.Query.CompiledQueryCache.Execute[TResult](Expression
 query, DataStore dataStore, QueryContext queryContext) in D:\k\EntityFramework\
src\EntityFramework.Core\Query\CompiledQueryCache.cs:line 54
   at Microsoft.Data.Entity.Query.EntityQueryProvider.Execute[TResult](Expressio
n expression) in D:\k\EntityFramework\src\EntityFramework.Core\Query\EntityQuery
Provider.cs:line 59
   at System.Linq.Queryable.Any[TSource](IQueryable`1 source)
   at Repro1251.Program.Main(String[] args) in D:\Projects\Repro1251\Repro1251\Program.cs:line 38

**Comment by maumar** [comment]
note that: ctx.Customers.Where(c => c.Id == 1).OrderBy(c => c.Id).Any() works fine

**Comment by DiegoAlejandroGomez** [comment]
You can reproduce it using the following test on QueryTestBase.cs

[Fact]
        public virtual void OrderThenWhere()
        {
            AssertQuery<Customer>(
                cs => cs.OrderBy(c => c.CustomerID).Where(c => c.CustomerID != string.Empty).Any());
        }

**Comment by maumar** [comment]
fixed in 5ca3664a766edd2d47c04efbe03d7d1d78c12a83 and 1f8938316864038a69d9ec1b28a239f5dd28799e
